server/auth.js

```diff
@@ -1,137 +1,122 @@
+        try {
+          if (!decoded) {
+            return { isValid: false };
+          }
+          const header = req.headers.authorization;
+          if (!header || !header.indexOf('Bearer ') === 0) {
+            return { isValid: false };
+          }
+          const token = header.split(' ')[1];
+          const isApiRequest = decoded && decoded.payload && decoded.payload.iss === `https://${config('AUTH0_DOMAIN')}/`;
+          const isDashboardAdminRequest = decoded && decoded.payload && decoded.payload.iss === config('PUBLIC_WT_URL');
+
+          if (isApiRequest) {
+            if (decoded.payload.gty && decoded.payload.gty !== 'client-credentials') {
+              return { isValid: false };
             }
 
-            return jwt.verify(token, key, jwtOptions.resourceServer.verifyOptions, (err) => {
-              if (err) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
-
-              if (decoded.payload.gty && decoded.payload.gty !== 'client-credentials') {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
-
-              if (!decoded.payload.sub.endsWith('@clients')) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
-
-              if (decoded.payload.scope && typeof decoded.payload.scope === 'string') {
-                decoded.payload.scope = decoded.payload.scope.split(' '); // eslint-disable-line no-param-reassign
-              }
+            if (!decoded.payload.sub.endsWith('@clients')) {
+              return { isValid: false };
+            }
 
-              return callback(null, true, decoded.payload);
+            const client = jwksRsa({
+              jwksUri: `https://${config('AUTH0_DOMAIN')}/.well-known/jwks.json`
             });
-          });
-        } else if (decoded && decoded.payload && decoded.payload.iss === config('PUBLIC_WT_URL')) {
-          return jwt.verify(
-            token,
-            jwtOptions.dashboardAdmin.key,
-            jwtOptions.dashboardAdmin.verifyOptions,
-            (err) => {
-              if (err) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
-
-              if (!decoded.payload.access_token || !decoded.payload.access_token.length) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
+            const getKey = (jwtHeader, cb) => {
+              client.getSigningKey(jwtHeader.kid, (err, key) => {
+                if (err) {
+                  return cb(err);
+                }
+                const signingKey = key.publicKey || key.rsaPublicKey;
+                cb(null, signingKey);
+              });
+            };
+
+            return new Promise((resolve) => {
+              jwt.verify(
+                token,
+                getKey,
+                {
```

Comment:
I was expecting to see `jwtOptions.resourceServer.verifyOptions`  used here. Any reason you are deviating from the [current configuration](https://github.com/auth0-extensions/auth0-account-link-extension/pull/198/files#diff-66454a9901457354fbdee0691510d38f0dd8f9b2c85c7cf31961ba738aa66ce7L54-L59) for JWT verification? Seems like the key is being loaded a bit differently as well

Reply:
The `hapiJwt2Key` function wasn't working as expected. I just followed the docs with the regular `getSigningKey` callback function. I also passed in the verify options directly.

Reply:
I guess when using `jwks` and dynamic keys can cause issues if you don't use callbacks and try and use `promisify` it

Reply:
I'll try this again though with `hapiJwt2Key` and using async promisify